### PR TITLE
Add Cleanlab's LLM integration

### PR DESCRIPTION
**Pull request by AshishSardana**
# Description

This PR integrates [Cleanlab's Trustworthy Language Model (TLM)](https://cleanlab.ai/tlm/) to llama-index via [cleanlab-studio](https://github.com/cleanlab/cleanlab-studio) library. The class provides a structured way to send completion requests to TLM, with support for both synchronous and asynchronous calls, as well as streaming responses.

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by ravi03071991** [comment]
@AshishSardana thanks for the integration. Could you please add an example notebook similar to the [existing notebooks](https://github.com/run-llama/llama_index/tree/main/docs/docs/examples/llm)?

Please add `tests`, you check [existing integrations](https://github.com/run-llama/llama_index/tree/main/llama-index-integrations/llms/llama-index-llms-anthropic/tests) for reference.

**Comment by AshishSardana** [comment]
Team @ravi-theja, @logan-markewich 

When executing `llm.complete(prompt)`, I'm facing the error -
```ValueError: Cannot use llm_completion_callback on an instance without a callback_manager attribute.```

Can you please suggest what I'm doing wrong?

File: llama-index-integrations/llms/llama-index-llms-cleanlab/llama_index/llms/cleanlab/base.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        api_key: Optional[str] = None,
+        quality_preset: Optional[str] = "medium",
+        additional_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """Initialize params."""
+        additional_kwargs = additional_kwargs or {}
+
+        api_key = get_from_param_or_env("api_key", api_key, "CLEANLAB_API_KEY")
+        #self.quality_preset = quality_preset
+
```
Comment (ravi03071991) [inline llama-index-integrations/llms/llama-index-llms-cleanlab/llama_index/llms/cleanlab/base.py:37]: You should include `CallbackManager`. You can find a reference [here](https://github.com/run-llama/llama_index/blob/554bfaef06e9784354e7dc46cf1c6d9c97985845/llama-index-integrations/llms/llama-index-llms-llama-api/llama_index/llms/llama_api/base.py#L76).

Comment (AshishSardana) [reply llama-index-integrations/llms/llama-index-llms-cleanlab/llama_index/llms/cleanlab/base.py:45]: thanks!

**Review by ravi03071991 — COMMENTED** [review]
Looks good. Please take care of linting and other unit tests before merge.

File: llama-index-integrations/llms/llama-index-llms-cleanlab/tests/test_llms_cleanlab.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from llama_index.core.base.llms.base import BaseLLM
+from llama_index.llms.cleanlab import CleanlabTLM
+
+
+def test_embedding_class():
```
Comment (ravi03071991) [inline llama-index-integrations/llms/llama-index-llms-cleanlab/tests/test_llms_cleanlab.py:5]: it should be `test_llms_cleanlab`

**Comment by ravi03071991** [comment]
@AshishSardana `BUILD` files are missing in tests and other folders. Please look into other LLM integrations for reference.

File: llama-index-integrations/llms/llama-index-llms-cleanlab/llama_index/llms/cleanlab/BUILD
Comment (AshishSardana) [inline llama-index-integrations/llms/llama-index-llms-cleanlab/llama_index/llms/cleanlab/BUILD:1]: @ravi03071991 is this correct? i didn't create or modify this BUILD file, hence double checking.

**Comment by AshishSardana** [comment]
> Looks good. Please take care of linting and other unit tests before merge.

I ran `make format; make lint` and `pytest()` to confirm every check is passed.

**Comment by AshishSardana** [comment]
@ravi03071991 can you see why these 2 checks are failing? the unit tests pass locally.
image

**Comment by ravi03071991** [comment]
@AshishSardana take care of Linting and then we are good to go.

File: docs/docs/examples/llm/cleanlab.ipynb [hunk 3]
```diff
@@ -0,0 +1,193 @@
+{
```
Comment (masci) [inline docs/docs/examples/llm/cleanlab.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    llm = CleanlabTLM(
</pre><p>Can you compact these three lines into a single one: <code>llm = CleanlabTLM(api_key="your_api_key")</code>? The linter is complaining  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/13994/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Review by masci — APPROVED** [review]
Piiggybacking on Ravi's review, LGTM!

**Pull request merged**